Master

SugarF0x opened the pull request.

SugarF0x committed:
:blue_book: Add terms with completion checkboxes

SugarF0x committed:
:corn: Remove IDE settings from git

I always keep forgetting these are auto-included

SugarF0x committed:
:white_check_mark: Setup basic API fetch

> Fetch JSON via proxy to comply with CORS
> Display fetched data in dummy div

SugarF0x committed:
:art: Add dummy text display

> Trim description down to first paragraph with no tags
> Display said paragraph in card
> Display post title
> Outline and margin out cards

SugarF0x committed:
:blue_book: Update docs' TODO

Jobs display - check!

SugarF0x committed:
:white_check_mark: Add heart icon

> Add heart icon to the bottom right of post card
> Link icon's color to post.state.fav

SugarF0x committed:
:white_check_mark: Add heart icon click functionality

> Change state (and color) on click

SugarF0x committed:
:white_check_mark: Add favorites' saving to localStorage

> Save post's ID to localStorage

SugarF0x committed:
:white_check_mark: Add favorites' pull from localStorage

> Pull favorites' array from localStorage
> Assign favorite state to IDs from said array

SugarF0x committed:
:corn: Remove indev data placeholder

> Remove hardwired data
> Uncomment data fetch

SugarF0x committed:
:milky_way: Complete

ghost commented:
Кажется, что не до конца было понято или согласовано тз

Хотелось бы видеть более сложное приложение, где по поисковому запросу выдавались вакансии. Это можно сделать с этим API (поисковой запрос - description, город - location)

Тогда и для избранных вакансий должен быть отдельный блок/страница

Желательно использование менеджера состояний (redux предпочтительнее, так как его мы используем в наших проектах)

SugarF0x commented:
Будет доделано

SugarF0x committed:
:blue_book: Add additional tasks and Commit syntax

The terms were incomplete, now have to finish with the missing tasks

SugarF0x committed:
:white_check_mark: Setup Redux and configure skeleton

> Connect Redux to project
> Define data stored

SugarF0x committed:
:recycle: Move Post to a separate component

SugarF0x committed:
:recycle: Cleanup code style

SugarF0x committed:
:white_check_mark: Add data fetch status field to Redux

SugarF0x committed:
:white_check_mark: Add custom data fetch via input field

> Call GitHub Jobs API with parameters specified in search fields
  * description
  * location
> Display all the fetch stages

SugarF0x committed:
:white_check_mark: Add 404 fetch status option

> Display 404 text when API call returned empty array

SugarF0x committed:
:blue_book: Update README Terms completion

Two points have been introduces:
  * Search bar
  * Redux

SugarF0x committed:
:art: Style dummy data button with material UI solution

SugarF0x committed:
:white_check_mark: Add favorite functionality

> Manage favorites' states through Redux action
> Manage localStorage through Redux action

SugarF0x committed:
:white_check_mark: Add favorites only display

> A button beneath search bar will only fetch favorites

SugarF0x committed:
:art: Add spinner during fetch phase

SugarF0x committed:
:milky_way: Done

SugarF0x committed:
:blue_book: Add new Terms

The project is fine for the most part,
  but newly added tasks would be much welcome

SugarF0x committed:
:blue_book: Document failed eslint airbnb introduction attempt

Well, was worth a shot

SugarF0x committed:
:recycle: Fix onClick functions (some of them)

Some functions were considered by TypeScript as unassignable
This will not be an issue with regular js, so I'l leave it here for now

SugarF0x committed:
:recycle: Fix render inline conditionals

> Move inline checks from render to a separate method

SugarF0x committed:
:recycle: Remove unnecessary constructor

SugarF0x committed:
:blue_book: Check for stateless classes

> I was to refactor stateless classes into functions
  but found none, so will leave it as is

SugarF0x committed:
:white_check_mark: Add timestamp display

> Display posts' date and time

SugarF0x committed:
:white_check_mark: Add filter to memory to prevent search spam

> Disable button when filter input matches posts filter

SugarF0x committed:
:hammer_and_wrench: Broken favorites data mutation

Recent changes to data storage broke favorites display
  since they did not include posts as separate object key
  nor did they pass any filter

> Pass favorites' posts as separate mutation key
> Pass favorites' filter as a dummy plug filter
  * This is a temporary solution since favorites will
    be turned into separate posts-like data object

SugarF0x committed:
:recycle: Convert favorites' data to a posts-like data instance

> Store IDs in said instance

Actual posts storage as well as fetching is still to be finihsed

SugarF0x committed:
:white_check_mark: Add full posts storage to avoid refetch

> Store posts in separate search key
> Store favorites in sepearte favs key
> Use said keys to display data respectively
> Call said data on call from store rather from API

SugarF0x committed:
:recycle: Introduce Thunk

> Move in-component fetching to a thunk-action

SugarF0x committed:
:recycle: Prettify code

Used by own style but tried to have it resemble airbnb
  as much as possible
Newer projects will have actual airbnb included

SugarF0x committed:
:milky_way: Done yet again

liyagan commented on src/components/Post.tsx:
```diff
@@ -0,0 +1,86 @@
+  formatDate(str: string): string {
+    let date = new Date(str);
+
+    let day   = date.getDay().toString().length === 1 ? `0${ date.getDay() }` : date.getDay();
+    let month = date.getMonth().toString().length === 1 ? `0${ date.getMonth() }` : date.getMonth();
+    let year  = date.getFullYear();
+    let hour  = date.getHours().toString().length === 1 ? `0${ date.getHours() }` : date.getHours();
+    let mins  = date.getMinutes().toString().length === 1 ? `0${ date.getMinutes() }` : date.getMinutes();
```
Можно  добавить  функцию форматтер

liyagan commented on src/components/Posts.tsx:
```diff
@@ -0,0 +1,88 @@
+          <Button style={ {margin: '1rem'} }
+                  variant="contained"
+                  color="primary"
+                  onClick={ () => {
```
Не обязхательно создавать новую функцию, можно просто передать `onClick={getDummyPosts}`
